lambda_functions/healthcheck/healthcheck.py

```diff
@@ -0,0 +1,89 @@
+import logging
+
+import boto3
+import requests
+
+# Configure logging
+logger = logging.getLogger()
+logger.setLevel(logging.ERROR)
+
+def check_web_url_health(url, expected_string, env):
+    """
+    check_web_url_health is a function that calls the url from the event
+    and checks if the status code is 200 and the expected string exists on that url.
+
+    :param url: Url of url to be checked
+    :type url: string
+    :param expected_string: String of text to be checked
+    :type expected_string: String
+    :param env: This is for us to distinguish the message between the dev, preprod and prod environments
+    :type env: string
+    :raises logger.error: will log an error to the lambda log group
+    :raises logger.error: variable
+    """
+
+    # Ping the url
+    response = requests.get(url, timeout=5)
+
+    # Define metric data
+    cloudwatch = boto3.client('cloudwatch')
+    metric_data = cloudwatch.put_metric_data(
+        MetricData = [
+            {
+                'MetricName': f'{env}-response',
+                'Dimensions': [
+                    {
+                    'Name': 'EndpointUrl',
+                    'Value': '1',
+                    },
+                    {
+                    'Name': 'ExpectedString',
+                    'Value': '1',
+                    }
+                ],
+                'Unit': 'None',
+                'Value': 1
+            },
+        ],
+        Namespace = 'SML-Healthcheck.'
+    )
+
+    # If the response code is not 200 or the response text does not 
+    # contain the expected string then we log an error and fail the lambda
+    if response.status_code != 200:
+        logger.info(f"Metric Data: , {metric_data}")
+        raise logger.error(f"Error: Status code expected to be 200 but is {response.status_code}")
```

Comment:
This doesn't look right, raise is usually used to raise an exception - you are using it with logger.error which seems wrong. Can you take another look at this.

Reply:
The aim is to fail the lambda here otherwise we do not get an alarm trigger. We need to raise an error to do this if response status code is not 200.

Reply:
I have changed it to Exception